pkg/server/plugin/notifier/k8sbundle/k8sbundle_watcher.go

```diff
@@ -62,17 +64,21 @@ func (b *bundleWatcher) watch(ctx context.Context) error {
 }
 
 // watchEvent triggers the read-modify-write for a newly created object
-func (b *bundleWatcher) watchEvent(ctx context.Context, client kubeClient, event watch.Event) (err error) {
+func (b *bundleWatcher) watchEvent(ctx context.Context, client kubeClient, event watch.Event) error {
 	if event.Type == watch.Added {
 		objectMeta, err := meta.Accessor(event.Object)
 		if err != nil {
 			return err
 		}
 
-		b.p.log.Debug("Setting bundle for new object", "name", objectMeta.GetName())
-		if err = b.p.updateBundle(ctx, b.config, client, objectMeta.GetNamespace(), objectMeta.GetName()); err != nil {
+		err = b.p.updateBundle(ctx, b.config, client, objectMeta.GetNamespace(), objectMeta.GetName())
+		if err != nil && status.Code(err) != codes.FailedPrecondition {
+			// Ignore FailPrecondition errors for when SPIRE is booting and we receive an event prior to
+			// IdentityProvider being initialized. In this case the BundleLoaded event will come
+			// to populate the caBundle, so its safe to ignore this error.
 			return err
 		}
+		b.p.log.Debug("Set bundle for object", "name", objectMeta.GetName(), "event", event.Type)
```

Comment:
This requires a slight reordering otherwise we'll log that we set the bundle when we encounter FailedPrecondition, maybe something like:
```
err = b.p.updateBundle(ctx, b.config, client, objectMeta.GetNamespace(), objectMeta.GetName())
switch {
case err == nil:
	b.p.log.Debug("Set bundle for object", "name", objectMeta.GetName(), "event", event.Type)
case status.Code(err) == codes.FailedPrecondition:
	// Ignore FailPrecondition errors for when SPIRE is booting and we receive an event prior to
	// IdentityProvider being initialized. In this case the BundleLoaded event will come
	// to populate the caBundle, so its safe to ignore this error.
default:
    return err
}```